plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/UpdateNuageVspDeviceCommand.java

```diff
@@ -0,0 +1,42 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+package com.cloud.agent.api;
+
+import java.util.Map;
+
+public class UpdateNuageVspDeviceCommand extends Command {
+
+    Map<String, String> _parametersToBeUpdated;
+
+    public UpdateNuageVspDeviceCommand(Map<String, String> parametersToBeUpdated) {
+        super();
+        this._parametersToBeUpdated = parametersToBeUpdated;
+    }
+
+    public Map<String, String> getParametersToBeUpdated() {
+        return _parametersToBeUpdated;
```

Comment:
A defensive copy or the value wrapped in a call to  ``Collections.unmodifiableMap(...)``should be returned here in order to avoid unintended side effects by users of the return value.